src/main/java/com/solr/clientwrapper/rest/SolrSearchOperationResource.java

```diff
@@ -0,0 +1,115 @@
+package com.solr.clientwrapper.rest;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.SolrQuery.SortClause;
+import org.apache.solr.client.solrj.beans.DocumentObjectBinder;
+import org.apache.solr.client.solrj.impl.HttpSolrClient;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.SolrDocumentList;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.solr.clientwrapper.infrastructure.solrbean.SolrCollectionIndex;
+import com.solr.clientwrapper.infrastructure.solrbean.SolrSearchResult;
+
+@RestController
+@RequestMapping("/solr")
+public class SolrSearchOperationResource {
+	/*
+	 * /////// ############## Solr Search Operation ################ /////////
+	 * Author: Piyush Ojha; NeoSOFT Tech.
+	 */    
+	
+    private static final String SOLR_DATA_NAME = "techproducts";
+    private static final String SOLR_DATA_NAME_DEFAULT = "techproducts";
+    
+    @Autowired
+    private SolrClient solrClient;
+	@Autowired
+	SolrSearchResult solrSearchResult;
+	@Autowired
+	SolrCollectionIndex solrCollectionIndex;
+
+
+	@GetMapping(value = "/search")
+	public ResponseEntity<SolrSearchResult> searchRecordsInGivenCollection(
+											@RequestParam(defaultValue = SOLR_DATA_NAME_DEFAULT) String collection, 
+											@RequestParam(defaultValue = "name") String queryField, 
+											@RequestParam(defaultValue = "*") String searchTerm,
+											@RequestParam(defaultValue = "0") String startRecord, 
+											@RequestParam(defaultValue = "5") String pageSize, 
+											@RequestParam(defaultValue = "id") String tag, 
+											@RequestParam(defaultValue = "asc") String order)
+													throws SolrServerException, IOException {
+		SolrClient client = new HttpSolrClient.Builder("http://localhost:8983/solr/"+collection).build();
```

Comment:
Solr adaptor should be used to call any solr api(karthik is creating one use that once its done)

Reply:
I have shifted solr api call to service layer now, just as Karthik has done it. Please see if it's fine now.